core/src/dbs/processor.rs

```diff
@@ -147,8 +165,11 @@ impl<'a> Processor<'a> {
 						self.process_range(stk, ctx, opt, stm, &tb, v).await?
 					}
 				}
-				Iterable::Table(v, keys_only) => {
+				Iterable::Table(v, mut keys_only) => {
 					let ctx = Self::check_query_planner_context(ctx, &v);
+					if keys_only {
+						keys_only = Self::check_keys_only_permissions(&ctx, opt, stm, &v).await?;
+					}
```

Comment:
Should we also be adding this to line 161/162?

Reply:
Yes, I was considering this. But then I wrote the test checking ranges. And for some reason, the test are passing.

https://github.com/surrealdb/surrealdb/blob/001011cee458c27db7a72062c6530408771b76b1/sdk/tests/group.rs#L1013-L1025

I am investigating "some reasons", bear with me...

Reply:
So this issue is, "in fine", a false positive. Security is already enforced by this existing mechanism:

https://github.com/surrealdb/surrealdb/blob/439ab99e15314ec5cf87047bf58246db646e3f8c/core/src/doc/select.rs

My initial test did not correctly set the permissions, so I initially thought we had an issue. But we don't, permissions are perfectly checked!

We can merge this PR to have a test that checks it and prevents any regression.